src/txorphanage.cpp

```diff
@@ -138,32 +212,38 @@ void TxOrphanage::LimitOrphans(unsigned int max_orphans)
         if (nErased > 0) LogPrint(BCLog::TXPACKAGES, "Erased %d orphan tx due to expiration\n", nErased);
     }
     FastRandomContext rng;
-    while (m_orphans.size() > max_orphans)
+    while (m_orphans.size() > max_orphans || m_total_orphan_bytes > max_total_size)
     {
         // Evict a random orphan:
         size_t randompos = rng.randrange(m_orphan_list.size());
-        EraseTxNoLock(m_orphan_list[randompos]->first);
+        const auto& wtxid = m_orphan_list[randompos]->second.tx->GetWitnessHash();
+        expired_and_evicted.emplace_back(wtxid);
+        EraseTxNoLock(wtxid);
         ++nEvicted;
```

Comment:
we should `+=` like the above call or do `++` for both I think

Reply:
done